src/core/observer/dep.js

```diff
@@ -36,6 +37,12 @@ export default class Dep {
   notify () {
     // stabilize the subscriber list first
     const subs = this.subs.slice()
+    if (process.env.NODE_ENV !== 'production' && !config.async) {
```

Comment:
Is it necessary to check for `NODE_ENV` ? I usually have different tests configurations, and one of them sets `NODE_ENV` to production to be as close as possible to the production behaviour. By reading this code, I guess my tests won't work.